### PR TITLE
[WIP] Support new AMP unified API from PyTorch 19.04 NGC container

### PR DESCRIPTION
**Pull request by pkashinkunti**
- Update Dockerfile to use 19.04 PyTorch NGC container
- Transition from old FP16 optimizer from APEX to new AMP unified API
- Make model export to TRT no longer use ONNX upsample workaround by default
- Update `setup.py` to compile for multiple supported GPU architectures

`test.sh` script passes all tests with this update.

TODO:
--Verify training & inference accuracy from scratch and with pretrained models

### CODE REVIEW

File: retinanet/train.py [hunk 1]
```diff
@@ -16,28 +16,27 @@
...
 
-    # Prepare dataset
-    if verbose: print('Preparing dataset...')
-    data_iterator = (DaliDataIterator if use_dali else DataIterator)(
-        path, jitter, max_size, batch_size, model.stride,
-        world, annotations, training=True)
-    if verbose: print(data_iterator)
-
     # Prepare model
     nn_model = model
```
Comment (florischabert) [inline retinanet/train.py]: You can remove this if we're not using it anymore

Comment (pkashinkunti) [reply retinanet/train.py]: nn_model is still used here for saving the model checkpoint (without the DDP wrapper): https://github.com/NVIDIA/retinanet-examples/pull/20/files/780ca6e9f989ce88b3bdaca88f7784066362eb12#diff-9339c352aabfb452a77c9714dac89004R146

Comment (pkashinkunti) [reply retinanet/train.py]: Also, just did a test for resuming training and it appears to work fine.

**Comment by florischabert** [comment]
:+1:
